### PR TITLE
feat(linter): implement `eslint/no-invalid-regexp`

### LINKED ISSUE #611: [$50 Bounty] feat(linter): eslint/no-invalid-regexp
Implement ESLint's [`no-invalid-regexp`](https://eslint.org/docs/latest/rules/no-invalid-regexp#rule-details) rule.

To get started, run
```sh
just new-rule no-invalid-regexp
```

### PR DESCRIPTION
**Pull request by Boshen**
closes #611

@leaysgur Kicking of `no-invalid-regexp`, feel free to take over and claim the bounty on #611 :grin: 

I can continue if you wanna work on other stuff (getting confused in prettier land ;-))

### CODE REVIEW

File: crates/oxc_linter/src/rules/eslint/no_invalid_regexp.rs [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+        ctx.diagnostic(diagnostic);
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/eslint/no_invalid_regexp.rs:95]: Some of these may have incorrect `\`, please double check with https://github.com/eslint/eslint/blob/main/tests/lib/rules/no-invalid-regexp.js

**Comment by leaysgur** [comment]
OK, I will take this. 🤭 (Prettier... too far to go...)

**Comment by leaysgur** [comment]
Basically done.

But I'd like to fix diagnostic labels in the parser to align display.

**Comment by leaysgur** [comment]
Benchmark...why... 😮

**Comment by leaysgur** [comment]
...🤷🏻‍♂️ 

image

**Comment by Boshen** [comment]
Strange ... I'll take a look.

**Comment by leaysgur** [comment]
According to https://github.com/oxc-project/oxc/pull/5570#issuecomment-2335165856 , is it because `New|CallExpression` is called as frequently as `UpdateExpression`?

But the commit just before the performance problem was 1e26fce .

image

And here is a diff of that commit and the latest.

https://github.com/oxc-project/oxc/compare/1e26fce...6a8c379

Since I only adjusted `Span`, why were there no performance issues at this point...?

**Comment by Boshen** [comment]
Codspeed is drunk, I don't see anything wrong.

**Pull request merged**